frontend/src/App.js

```diff
@@ -39,16 +51,16 @@ function App() {
           <Route path="profile">
             <Route index element={<Profile />} />
             <Route path=":id" element={<Profile />} />
-            <Route path="edit" element={<ProfileEdit />} />
+            <Route path="edit/:id" element={<ProfileEdit />} />
             <Route path="*" element={<h1>404 page not found</h1>} />
           </Route>
 
           <Route path="*" element={<h1>404 page not found</h1>} />
         </Route>
 
         {/* Auth paths */}
-        <Route path="signup" element={<SignUp />} />
-        <Route path="login" element={<SignIn />} />
+        <Route path="signup" element={<SignUp setToken={setToken} />} />
```

Comment:
Here you are passing props that you don't use in the component. I would remove it. Same for the SignUp comp